jetty-annotations/src/main/java/org/eclipse/jetty/annotations/AnnotationParser.java

```diff
@@ -561,31 +564,27 @@ public AnnotationParser(int javaPlatform, int asmVersion)
      */
     public void addParsedClass (String classname, Resource location)
     {
-        List<String> list = new ArrayList<>(1);
-        if (location != null)
-            list.add(location.toString());
-
-        List<String> existing = _parsedClassNames.putIfAbsent(classname, list);
+        String locationStr = location.toString();
+        String existing = _parsedClassNames.putIfAbsent(classname, locationStr);
+        
         if (existing != null)
-        {
-            existing.addAll(list);
-            LOG.warn("{} scanned from multiple locations: {}", classname, existing);
-        }
+            LOG.warn("{} scanned from multiple locations: {}, {}", classname, existing, locationStr);
```

Comment:
Revert back to log `location`, not `locationStr`.

Reply:
But that's 2 toStrings! One for the autobox to String on the putIfAbsent, and then the other on location in the LOG.warn.

Reply:
OK ok. You win. Change made.